docs/in-depth/new-module.md

```diff
@@ -0,0 +1,63 @@
+# Adding a new module
+
+You can easily extend MoMEMta by adding new modules to perform operations not supported by the base modules.
+
+## Defining the module's interface
+
+You declare a new module in MoMEMta by subclassing the [Module class](https://momemta.github.io/MoMEMta/dev/classModule.html), and then registering the new module into the MoMEMta system.
+
+Let's suppose you want to declare a new module printing `Hello world!` at each step of the integration. You first need to include the right header:
+
+```C++
+#include <momemta/Module.h>
+```
+
+and then to subclass the `Module` class:
+
+```C++
+class HelloWorldModule: public Module {
+
+};
+```
+
+If you want your new module to be useful, you'll want to override some specific functions, starting by the constructor and the  [work](https://momemta.github.io/MoMEMta/dev/classModule.html#a43c0b7c70e9dd8b0cf6c78aa5052eeee) function:
+
+```C++
+class HelloWorldModule: public Module {
+public:
+  HelloWorldModule(PoolPtr pool, const ParameterSet& parameters): Module(pool, parameters.getModuleName()) { }
+
+  virtual Status work() override {
+    std::cout << "Hello world!" << std::endl;
+  }
+};
+```
+
+The `work` function is called by MoMEMta each time the integrand needs to be evaluated (thousand times per integration). With this code, our module would only output the string `Hello world!` and not produce anything useful, but it's fine, that's what we wanted! The only remaining thing is to register the new module into the system.
+
+## Registering the module
+
+Before being available to MoMEMta, a module must be registered into the system. This is done using the `REGISTER_MODULE` macro. After the declaration of your module, simply add:
+
+```C++
+REGISTER_MODULE(HelloWorldModule)
+    .Sticky();
+```
+
+Because your module does not produce any output, MoMEMta will automatically remove it from the computation graph, since it does not contribute in any way to the final integrand value. You can prevent this behaviour by registering your module as `sticky`: a `sticky` module is not removed from the computation graph if it's found to not contribute.
+
+## Loading the module
+
+So, you have successfully defining a new module and registered it with the system. In order to use it during the computation of the weight, you first have to _load_ the shared library where your module is.
```

Comment:
Not really I know, but that fell more into "How i compile C++" than "How I use MoMEMta", I'm not sure how to document that